pkg/entrypoint/entrypointer.go

```diff
@@ -311,7 +307,8 @@ func (e Entrypointer) Go() error {
 			resultPath = e.ResultsDirectory
 		}
 		if err := e.readResultsFromDisk(ctx, resultPath, result.TaskRunResultType); err != nil {
-			logger.Fatalf("Error while handling results: %s", err)
+			slog.Error("Error while substituting step artifacts: ", slog.Any("error", err))
+			return err
```

Comment:
I'm curious why `log.Fatalf` is replaced with `logger.Fatalf` in other places, but why are `slog.Error` used in these two instances?

Reply:
Yeah good catch, I did use `log.Fatalf` but the issue was it was giving golang lint error as `log.Fatalf will exit, and `defer cancel()` will not run` hence I did use `slog.Error`